### ISSUE TITLE
Scrubbing Sensitive Data for Span

### ISSUE THREAD

**Issue by IlkhamGaysin** [getsentry/sentry-ruby#1823]
Hey guys, I've my Ruby on Rails application accepting [Data URI scheme](https://en.wikipedia.org/wiki/Data_URI_scheme) in params.
Sentry has the limits on payload size described [here](https://develop.sentry.dev/sdk/data-handling/#variable-size). Meanwhile sentry does not call `before_send` hook to filter payload for `TransactionEvent`. 
[Here](https://github.com/getsentry/sentry-ruby/blob/master/sentry-ruby/lib/sentry/client.rb#L124) is the condition 

I've filtered the Data URI string on `before_send` and `before_breadcrumb` which reduced payload size a bit but the problem still remains the same the sentry server returns 413 error. I've debuged andn found that spans are not filtered and has that Data URI string. 
My question is why sentry does not allow to filter sensitive data from spans? Would that be reasonable to implemet that?

**Comment by st0012** [comment]
You can use `Event processors` for this, like:

```rb
# place this in sentry.rb, after the Sentry.init block
Sentry.get_current_scope.add_event_processor do |event, _hint|
  if event.is_a?(Sentry::TransactionEvent)
    event.spans.delete_if do |span_hash|
      # span data in hash form
    end
  end

  event
end
```

But I think this is less than ideal and we should provide something similar to `before_send` but just for transactions. @sl0thentr0py

**Comment by IlkhamGaysin** [comment]
Do you need guys help? I can push a PR for this @sl0thentr0py @st0012

**Comment by sl0thentr0py** [comment]
@IlkhamGaysin for now you'll just have to use the event processor. Adding new APIs needs to go through a consensus process for having a [Unified API](https://develop.sentry.dev/sdk/unified-api/) across most of our SDKs. I personally do want to add something like this but we'll need to wait a bit for feedback from other language folks.

**Comment by st0012** [comment]
@sl0thentr0py Can I close this as not planned?

**Comment by sl0thentr0py** [comment]
@st0012 no, we're actually almost close to making a decision on this now. You can follow the process at 
https://github.com/getsentry/rfcs/pull/19

(Side note: we're trying to be more public about our decision making process so that's the new Sentry RFC repo where we do all our discussions. Feel free to pitch in!)

**Comment by sl0thentr0py** [comment]
we now have `before_send_transaction` so closing.
